flake.nix

```
@@ -0,0 +1,79 @@
+{
+  description = "A Rust port of tmux";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/24.11";
+    crane.url = "github:ipetkov/crane";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      crane,
+      flake-utils,
+      rust-overlay,
+      ...
+    }:
+    flake-utils.lib.eachDefaultSystem (
+      system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ (import rust-overlay) ];
+        };
+
+        inherit (pkgs) lib;
+
+        # use latest stable Rust compiler
+        craneLib = (crane.mkLib pkgs).overrideToolchain (p: p.rust-bin.stable.latest.default);
+
+        commonArgs = {
+          # include Rust sources, and anything else required by build
+          src = lib.fileset.toSource {
+            root = ./.;
+            fileset = lib.fileset.unions [
+              (craneLib.fileset.commonCargoSources ./.)
+              (lib.fileset.fileFilter (file: file.hasExt "lalrpop") ./.)
+              ./README.md
+            ];
+          };
+          strictDeps = true;
+        };
+
+        tmux-rs = craneLib.buildPackage (
+          commonArgs
+          // {
+            cargoArtifacts = craneLib.buildDepsOnly commonArgs;
+            cargoExtraArgs = "--verbose";
+
+            # libraries we link with (per build.rs)
+            buildInputs = [
+              pkgs.libevent.dev
+              pkgs.libtinfo.dev
```

```suggestion
              pkgs.libevent
              pkgs.ncurses
```

The ncurses package also has tinfo, so you don't need to pin to something way back in Feb. Specifying .dev is unnecessary.

---

Makes sense! It works with these changes, thanks :)